lib/iris/fileformats/nimrod_load_rules.py

```diff
@@ -25,58 +28,306 @@
+        field.source = "Nimrod pwind routine"
+    if field.source.strip() == "pwind":
+        field.source = "Nimrod pwind routine"
+
+    if getattr(field, "ensemble_member") == -98:
+        if "mean" not in field.title:
+            field.title = "mean_of_" + field.title
+        field.ensemble_member = field.int_mdi
+    if getattr(field, "ensemble_member") == -99:
+        if "spread" not in field.title:
+            field.title = "standard_deviation_of_" + field.title
+        field.ensemble_member = field.int_mdi
+
+    cube.rename(remove_unprintable_chars(field.title))
+
+
+def remove_unprintable_chars(input_str):
+    """
+    Removes unprintable characters from a string and returns the result.
+    """
+    return "".join(
+        c if c in string.printable else " " for c in input_str
+    ).strip()
 
 
 def units(cube, field):
     """
     Set the cube's units from the field.
 
+    Takes into account nimrod unit strings of the form unit*?? where the data
+    needs to converted by dividing by ??. Also converts units we know Iris
+    can't handle into appropriate units Iris can handle. This is mostly when
+    there is an inappropriate capital letter in the unit in the Nimrod file.
+    Some units still can't be handled by Iris so in these cases empty strings
+    are added as the cube's unit. The most notable unit Iris can't handle is
+    oktas for cloud cover.
+
     Unhandled units are stored in an "invalid_units" attribute instead.
 
     """
-    units = field.units.strip()
+    unit_exception_dictionary = {
+        "Knts": "knots",
+        "knts": "knots",
+        "J/Kg": "J/kg",
+        "logical": "1",
+        "Code": "1",
+        "mask": "1",
+        "mb": "hPa",
+        "g/Kg": "1",
+        "unitless": "1",
+        "Fraction": "1",
+        "index": "1",
+        "Beaufort": "",
+        "mmh2o": "kg/m2",
+        "n/a": "1",
+    }
+
+    field_units = remove_unprintable_chars(field.units)
+    if field_units == "m/2-25k":
+        # Handle strange visibility units
+        cube.data = (cube.data + 25000.0) * 2
+        field_units = "m"
+    if "*" in field_units:
+        # Split into unit string and integer
+        unit_list = field_units.split("*")
+        if "^" in unit_list[1]:
+            # Split out magnitude
+            unit_sublist = unit_list[1].split("^")
+            cube.data = cube.data / float(unit_sublist[0]) ** float(
+                unit_sublist[1]
+            )
+        else:
+            cube.data = cube.data / float(unit_list[1])
+        field_units = unit_list[0]
+    if "ug/m3E1" in field_units:
+        # Split into unit string and integer
+        unit_list = field_units.split("E")
+        cube.data = cube.data / 10.0 ** float(unit_list[1])
+        field_units = unit_list[0]
+    if "%" in field_units:
+        # Convert any percentages into fraction
+        unit_list = field_units.split("%")
+        if len("".join(unit_list)) == 0:
```

Comment:
What happens if this fails?
I guess it means we have something other than just "%" 
but the intention here, if that happens, is not clear  (to me, anyway!)

Reply:
I think this predates the `remove_unprintable_chars` method. We wouldn't have known whether the % was at the start, middle or end of the 8-char string. I've simplified the test.